src/modules/rlm_ldap/rlm_ldap.c

```diff
@@ -1025,16 +1077,21 @@ static unlang_action_t CC_HINT(nonnull) mod_authenticate(rlm_rcode_t *p_result,
 
 	RDEBUG2("Login attempt by \"%pV\"", &username->data);
 
-	/*
-	 *	Get the DN by doing a search.
-	 */
-	dn = rlm_ldap_find_user(inst, request, ttrunk, NULL, false, NULL, NULL, &rcode);
-	if (!dn) RETURN_MODULE_RCODE(rcode);
+	auth_ctx = talloc_zero(unlang_interpret_frame_talloc_ctx(request), ldap_auth_ctx);
+	*auth_ctx = (ldap_auth_ctx){
+		.password = password->vp_strvalue,
+		.thread = thread,
+		.inst = inst,
+		.ttrunk = ttrunk
+	};
 
 	/*
-	 *	Attempt a bind using the thread specific connection for bind auths
+	 *	Check for a cahed copy of the DN
 	 */
-	if (fr_ldap_bind_auth_async(request, thread, dn, password->vp_strvalue, true) < 0) RETURN_MODULE_FAIL;
+	rlm_ldap_find_user_cached(auth_ctx->dn, request);
```

Comment:
`rlm_find_user_dn_cached`... that confused me for a moment... if we're looking up a cached user why aren't we checking the return code :)